src/safeposix/syscalls/sys_calls.rs

```diff
@@ -123,18 +164,23 @@ impl Cage {
                     }
                 }
             } else {
+                // If the value is None - mimic the behavior in the child's mutex table
                 new_cv_table.push(None);
             }
         }
         drop(cvtable);
 
-        //construct new cage struct with a cloned fdtable
+        // Clone the file descriptor table in the child's Cage object
         let newfdtable = init_fdtable();
+        // Loop from 0 to maximum value of file descriptor index
         for fd in 0..MAXFD {
             let checkedfd = self.get_filedescriptor(fd).unwrap();
+            // Get the lock for the file descriptor
             let unlocked_fd = checkedfd.read();
             if let Some(filedesc_enum) = &*unlocked_fd {
+                // Check the type of the file descriptor
                 match filedesc_enum {
+                    // If the fd is linked to a file
```

Comment:
Can you elaborate here about what are we doing with this `File` type File Descriptor?